idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceManualRangeWithIndicesCallsInspection.kt

```diff
@@ -0,0 +1,132 @@
+
+            when {
+                fqName.matches(REGEX_RANGE_TO) -> visitRangeToExpression(expression, holder)
+                fqName.matches(REGEX_UNTIL) -> visitUntilExpression(expression, holder)
+                else -> return@expressionVisitor
+            }
+
+        }
+    }
+
+    private fun visitRangeToExpression(expression: KtExpression, holder: ProblemsHolder) {
+        if (expression.getArguments()?.second?.deparenthesize()?.isSizeMinusOne() != true) return
+
+        val constantValue = expression.getArguments()?.first?.constantValueOrNull()
+        val rightValue = (constantValue?.value as? Number)?.toInt() ?: return
+        if (rightValue != 0) return
+
+
+        holder.registerProblem(
+            expression,
+            "'rangeTo or the '..' call should be replaced with .indices call",
+            ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+            ReplaceManualRangeWithIndicesCallQuickFix()
+        )
+
+    }
+
+    private fun visitUntilExpression(expression: KtExpression, holder: ProblemsHolder) {
+        if (expression.getArguments()?.second?.deparenthesize()?.isSizeOrLength() != true) return
+
+        val constantValue = expression.getArguments()?.first?.constantValueOrNull()
+        val rightValue = (constantValue?.value as? Number)?.toInt() ?: return
+        if (rightValue != 0) return
+
+
+        holder.registerProblem(
+            expression,
+            "'rangeTo or the '..' call should be replaced with .indices call",
+            ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+            ReplaceManualRangeWithIndicesCallQuickFix()
+        )
+
+    }
+
+
+    private fun KtExpression.isSizeMinusOne(): Boolean {
+        if (this !is KtBinaryExpression) return false
+        if (operationToken != KtTokens.MINUS) return false
+
+        val leftValue = left?.isSizeOrLength() ?: return false
+        val constantValue = right?.constantValueOrNull()
+        val rightValue = (constantValue?.value as? Number)?.toInt() ?: return false
+        return rightValue == 1 && leftValue
+    }
+
+}
+
+class ReplaceManualRangeWithIndicesCallQuickFix : LocalQuickFix {
+    override fun getName() = "Replace with indices"
+
+    override fun getFamilyName() = name
+
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+        val element = descriptor.psiElement as KtExpression
+        val args = element.getArguments() ?: return
+        val second = (args.second as? KtBinaryExpression) ?: return
+        val array = (second.left as? KtDotQualifiedExpression)?.getLeftMostReceiverExpression() ?: return
+
+        element.replace(
+            KtPsiFactory(element).createExpressionByPattern(
+                "$0.indices",
+                array
+            )
+
+        )
+    }
+
+}
+
+
+private fun KtExpression.deparenthesize() = KtPsiUtil.safeDeparenthesize(this)
+
+private val REGEX_RANGE_TO = """kotlin.(Char|Byte|Short|Int|Long).rangeTo""".toRegex()
```

Comment:
I think maybe we only want to check `Int` here, because `listOf(1,2,3)[4L]` doesn't compile anyways